record-and-playback/core/lib/recordandplayback/generators/audio.rb

```diff
@@ -31,43 +31,76 @@ class AudioEvents
     def self.create_audio_edl(events, archive_dir)
       audio_edl = []
       audio_dir = "#{archive_dir}/audio"
+      audios = {}
+      active_audios = []
 
       initial_timestamp = BigBlueButton::Events.first_event_timestamp(events)
       final_timestamp = BigBlueButton::Events.last_event_timestamp(events)
 
       # Initially start with silence
       audio_edl << {
         :timestamp => 0,
-        :audio => nil
+        :audios => nil
       }
 
       # Add events for recording start/stop
-      events.xpath('/recording/event[@module="VOICE"]').each do |event|
+      events.xpath('/recording/event[@module="VOICE" or @module="bbb-webrtc-sfu"]').each do |event|
         timestamp = event['timestamp'].to_i - initial_timestamp
         case event['eventname']
         when 'StartRecordingEvent'
           filename = event.at_xpath('filename').text
           filename = "#{audio_dir}/#{File.basename(filename)}"
           audio_edl << {
             :timestamp => timestamp,
-            :audio => { :filename => filename, :timestamp => 0 }
+            :audios => [{ :filename => filename, :timestamp => 0 }]
           }
         when 'StopRecordingEvent'
           filename = event.at_xpath('filename').text
           filename = "#{audio_dir}/#{File.basename(filename)}"
-          if audio_edl.last[:audio] && audio_edl.last[:audio][:filename] == filename
+          if audio_edl.last[:audios] && audio_edl.last[:audios][0][:filename] == filename

```

Comment:
I recommend using `audio_edl.last.dig(:audios, 0, :filename)` here to avoid the need to check the condition first, and that will also prevent "undefined method \`[]'" errors if `audio_edl.last[:audios]` is an empty array.

Same comment applies to the `StopWebRTCDesktopShareEvent` handling.